lib/chef/resource/windows_audit_policy.rb

```diff
@@ -0,0 +1,226 @@
+                     "Detailed File Share",
+                     "Removable Storage",
+                     "Central Policy Staging",
+                     "Sensitive Privilege Use",
+                     "Non Sensitive Privilege Use",
+                     "Other Privilege Use Events",
+                     "Process Creation",
+                     "Process Termination",
+                     "DPAPI Activity",
+                     "RPC Events",
+                     "Plug and Play Events",
+                     "Token Right Adjusted Events",
+                     "Audit Policy Change",
+                     "Authentication Policy Change",
+                     "Authorization Policy Change",
+                     "MPSSVC Rule-Level Policy Change",
+                     "Filtering Platform Policy Change",
+                     "Other Policy Change Events",
+                     "User Account Management",
+                     "Computer Account Management",
+                     "Security Group Management",
+                     "Distribution Group Management",
+                     "Application Group Management",
+                     "Other Account Management Events",
+                     "Directory Service Access",
+                     "Directory Service Changes",
+                     "Directory Service Replication",
+                     "Detailed Directory Service Replication",
+                     "Credential Validation",
+                     "Kerberos Service Ticket Operations",
+                     "Other Account Logon Events",
+                     "Kerberos Authentication Service",
+                    ]
+      resource_name :windows_audit_policy
+
+      description "The windows_audit_policy resource allows for configuring system and per-user Windows advanced audit policy settings."
+
+      examples <<~DOC
+      **Set Logon and Logoff policy to "Success and Failure"**:
+
+      ```ruby
+      windows_audit_policy "Set Audit Policy for "Logon and Logoff" actions to "Success and Failure" do
+        sub_category   %w(Logon Logoff)
+        success        true
+        failure        true
+        action         :set
+      end
+      ```
+
+      **Set Credential Validation policy to "Success"**:
+
+      ```ruby
+      windows_audit_policy "Set Audit Policy for "Credential Validation" actions to "Success" do
+        sub_category   "Credential Validation"
+        success        true
+        failure        false
+        action         :set
+      end
+      ```
+      **Enable CrashOnAuditFail option**:
+
+      ```ruby
+      windows_audit_policy "Enable CrashOnAuditFail option" do
+        crash_on_audit_fail        true
+        action                     :set
+      end
+      ```
+
+      DOC
+
+      property :sub_category, [String, Array],
+        coerce: proc { |p| Array(p) },
+        description: "The audit policy subcategory, specified by GUID or name. Defaults to system if no user is specified.",
```

Comment:
The way this was written made it sound like a default value of `default` would be used, based on how the descriptions of properties are typically written.

```suggestion
        description: "The audit policy subcategory, specified by GUID or name. Applied system-wide if no user is specified.",
```